packages/url_launcher/url_launcher/test/link_test.dart

```diff
@@ -50,66 +50,62 @@ void main() {
 
     testWidgets('calls url_launcher for external URLs with blank target',
         (WidgetTester tester) async {
-      FollowLink followLink;
+      FollowLink? followLink;
 
       await tester.pumpWidget(Link(
         uri: Uri.parse('http://example.com/foobar'),
         target: LinkTarget.blank,
-        builder: (BuildContext context, FollowLink followLink2) {
+        builder: (BuildContext context, FollowLink? followLink2) {
           followLink = followLink2;
           return Container();
         },
       ));
 
-      when(mock.canLaunch('http://example.com/foobar'))
-          .thenAnswer((realInvocation) => Future<bool>.value(true));
-      clearInteractions(mock);
-      await followLink();
-
-      verifyInOrder([
-        mock.canLaunch('http://example.com/foobar'),
-        mock.launch(
-          'http://example.com/foobar',
+      mock
```

Comment:
Why do we need to migrate away from "when...` in mockito and introduce our own mocks? mockito already migrated to stable we should be able to just use it with nnbd, right?

Reply:
Sadly it's nowhere near that straightforward :( See https://github.com/dart-lang/mockito/blob/master/NULL_SAFETY_README.md

Reply:
Do we prefer auto-gen or manual? The doc mentioned autogen is recommended and I see that you manually generated the mocks.

Reply:
The Mockito generation adds a bunch more moving pieces; it didn't seem worth adding that extra complexity for a simple class with a couple of methods. There may be cases where we want to do it, but if it's simple to just start from `Fake` and build manually that seems easier.

This one is probably borderline since it's doing real mock validation rather than just faking; in a number of other cases we were using Mockito to make fakes, not actual mocks. We may want to give it a try for the remaining unit tests that need conversion and compare (or I can try it with this one if you'd like).

Reply:
I'll try with android_intent